lib/stb_image.ex

```diff
@@ -140,6 +139,24 @@ defmodule StbImage do
     StbImage.Nif.to_file(filename, extension, data, height, width, channels)
   end
 
+  @doc """
+  Encodes image to an in-memory binary.
+
+  ## Example
+
+      {:ok, buffer} = StbImage.to_memory(:png, img, height, width, channels)
+
+  """
+  def to_memory(extension, data, height, width, channels)
+      when is_binary(data) and is_integer(width) and width > 0 and is_integer(height) and
+             height > 0 and is_integer(channels) and channels > 0 do
+    if extension not in @to_file_exts do
```

Comment:
I'd rename this to `format` and do the same for `to_file`. `to_file` accepts an `:extension` option, but it's actually the format that we try to determine from the filename.